CMakeLists.txt

```diff
@@ -41,7 +41,7 @@ ELSE ()
     set(CMAKE_C_FLAGS "-D_GNU_SOURCE -std=gnu99 -fPIC ${CMAKE_C_FLAGS}")
     set(CMAKE_CXX_FLAGS "-std=c++11 -fno-rtti ${CMAKE_CXX_FLAGS}")
     set(CMAKE_C_FLAGS "-Wall -Werror ${CMAKE_C_FLAGS}")
-    set(CMAKE_CXX_FLAGS "-Wall -Wextra -Weffc++ ${CMAKE_CXX_FLAGS}")
```

Comment:
Is it not possible to set the c/cxx flags using FetchContent?

Else change the CMAKE_C_FLAGS / CMAKE_CXX_FLAGS only during FetchContent.

e.g.:
```cmake
set(SAVE_CXX ${CMAKE_CXX_FLAGS}
string(REPLACE "-Weffc++" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
FetchContent_Declare(
         googletest
         GIT_REPOSITORY https://github.com/google/googletest.git
         GIT_TAG release-1.8.1
         UPDATE_DISCONNECTED TRUE
         PREFIX ${CMAKE_BINARY_DIR}/gtest
         INSTALL_COMMAND ""
         GIT_TAG        release-1.10.0
 )
 FetchContent_MakeAvailable(googletest)
set(CMAKE_CXX_FLAGS ${SAVE_CXX})
unset(SAVE_CXX)
```

Not pretty, but then we can keep effc++ warning on

Reply:
The usage off effc++ is debatable, also CLang does not have it. I don't see a big problem in not having it. Wdyt?

Reply:
Personally I find effc++ one of the most important flags. Not always nice too see the warning after committing for OSX (clang), but almost always very useful warnings.

In other words I would like to keep effc++

Reply:
I've moved the inclusion of gtest to before the flags. This works. Since it is an external dependency, I think we can do without the flags. But.. if needed we can split up the flags (or even move Werror) so that the build does not fail due to gtest source.